tests/framework/resourcemanager.go

```diff
@@ -0,0 +1,313 @@
+
+			continue
+		}
+
+		if err := rm.K8sClient.Update(ctx, resource); err != nil {
+			return fmt.Errorf("error updating resource: %w", err)
+		}
+	}
+
+	return nil
+}
+
+// ApplyFromFiles creates or updates Kubernetes resources defined within the provided YAML files.
+func (rm *ResourceManager) ApplyFromFiles(files []string, namespace string) error {
+	ctx, cancel := context.WithTimeout(context.Background(), rm.TimeoutConfig.CreateTimeout)
+	defer cancel()
+
+	handlerFunc := func(obj unstructured.Unstructured) error {
+		obj.SetNamespace(namespace)
+		nsName := types.NamespacedName{Namespace: obj.GetNamespace(), Name: obj.GetName()}
+		fetchedObj := obj.DeepCopy()
+		if err := rm.K8sClient.Get(ctx, nsName, fetchedObj); err != nil {
+			if !apierrors.IsNotFound(err) {
+				return fmt.Errorf("error getting resource: %w", err)
+			}
+
+			if err := rm.K8sClient.Create(ctx, &obj); err != nil {
+				return fmt.Errorf("error creating resource: %w", err)
+			}
+
+			return nil
+		}
+
+		obj.SetResourceVersion(fetchedObj.GetResourceVersion())
+		if err := rm.K8sClient.Update(ctx, &obj); err != nil {
+			return fmt.Errorf("error updating resource: %w", err)
+		}
+
+		return nil
+	}
+
+	return rm.readAndHandleObjects(handlerFunc, files)
+}
+
+// Delete deletes Kubernetes resources defined as Go objects.
+func (rm *ResourceManager) Delete(resources []client.Object) error {
+	for _, resource := range resources {
+		ctx, cancel := context.WithTimeout(context.Background(), rm.TimeoutConfig.DeleteTimeout)
+		defer cancel()
+
+		if err := rm.K8sClient.Delete(ctx, resource); err != nil && !apierrors.IsNotFound(err) {
+			return fmt.Errorf("error deleting resource: %w", err)
+		}
+	}
+
+	return nil
+}
+
+// DeleteFromFile deletes Kubernetes resources defined within the provided YAML files.
+func (rm *ResourceManager) DeleteFromFiles(files []string, namespace string) error {
+	handlerFunc := func(obj unstructured.Unstructured) error {
+		obj.SetNamespace(namespace)
+		ctx, cancel := context.WithTimeout(context.Background(), rm.TimeoutConfig.DeleteTimeout)
+		defer cancel()
+
+		if err := rm.K8sClient.Delete(ctx, &obj); err != nil && !apierrors.IsNotFound(err) {
+			return err
+		}
+
+		return nil
+	}
+
+	return rm.readAndHandleObjects(handlerFunc, files)
+}
+
+func (rm *ResourceManager) readAndHandleObjects(
+	handle func(unstructured.Unstructured) error,
+	files []string,
+) error {
+	for _, file := range files {
+		data, err := rm.getFileContents(file)
+		if err != nil {
+			return err
+		}
+
+		decoder := yaml.NewYAMLOrJSONDecoder(data, 4096)
+		for {
+			obj := unstructured.Unstructured{}
+			if err := decoder.Decode(&obj); err != nil {
+				if errors.Is(err, io.EOF) {
+					break
+				}
+				return fmt.Errorf("error decoding resource: %w", err)
+			}
+
+			if len(obj.Object) == 0 {
+				continue
+			}
+
+			if err := handle(obj); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+// getContents takes a string that can either be a local file
```

Comment:
```suggestion
// getFileContents takes a string that can either be a local file
```
I think this is what you meant?